frontend/src/features/userProfile/UserProfile.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams, useHistory, useLocation } from 'react-router-dom'
+import queryString from 'query-string'
+
+import {
+  loadUserProfile,
+  loadUserProfileDatasets,
+  loadUserProfileFollowing
+} from './state/userProfileActions'
+import {
+  selectUserProfile,
+  selectUserProfileDatasets,
+  selectUserProfileFollowing,
+} from './state/userProfileState'
+import ContentBox from '../../chrome/ContentBox'
+import NavBar from '../navbar/NavBar'
+import Footer from '../footer/Footer'
+import {
+  UserProfileParams,
+  NewUserProfileParams,
+  CleanUserProfileParams
+} from '../../qri/userProfile'
+import { ContentTabs, Tab } from '../../chrome/ContentTabs'
+import UserProfileHeader from './UserProfileHeader'
+import UserProfileDatasetList from './UserProfileDatasetList'
+
+interface UserProfileProps {
+  path?: '/' | '/following'
+}
+
+const UserProfile: React.FC<UserProfileProps> = ({ path = '/' }) => {
+  const dispatch = useDispatch()
+  const history = useHistory()
+
+  const profile = useSelector(selectUserProfile)
+
+  const paginatedDatasetResults = useSelector(selectUserProfileDatasets)
+  const paginatedFollowingResults = useSelector(selectUserProfileFollowing)
+
+  const scrollContainer = useRef<HTMLDivElement>(null)
+  const { username: usernameParam } = useParams();
+
+  const { search } = useLocation()
+  const userProfileParams: UserProfileParams = NewUserProfileParams(queryString.parse(search))
+  const { page, sort } = userProfileParams
+
+  // if the query string ever changes, fetch new data
+  useEffect(() => {
+    dispatch(loadUserProfile(usernameParam))
+    dispatch(loadUserProfileDatasets(usernameParam, userProfileParams))
+    dispatch(loadUserProfileFollowing(usernameParam, userProfileParams))
+  }, [usernameParam, page, sort, dispatch])
+
+  // merges new query params with existing params, updates history
+  const updateQueryParams = (newQueryParams: Record<string, any>) => {
```

Comment:
again, like the pattern of pulling all the mutations into one function, makes for easy reading/grok-ing!

Reply:
:)